Algebra.java

```diff
@@ -25,43 +25,141 @@ public static void main(String args[]) {
 
 	// Returns x1 + x2
 	public static int plus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		int y1 = x1;
+		int y2 = x2;
+		if (y2 < 0) {
+			for (int i = 0; i > y2; i--) {
+				y1 --;
+			}	
+		} else {
+		for (int i = 0; i < y2; i++) {
+			y1 ++;
+		}
+	}
+		return y1;
 	}
 
 	// Returns x1 - x2
 	public static int minus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		int y1 = x1;
+		int y2 = x2;
+		if (y2 <0) {
+			for (int i = 0; i > y2; i--) {
+				y1 ++;
+			}
+		} else {
+			for (int i = 0; i < y2; i++) {
+			y1 --;
+		}
+	}
+		return y1;
 	}
 
 	// Returns x1 * x2
 	public static int times(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		int y1 = (x1 < 0) ? -x1 : x1;
+		int y2 = (x2 < 0) ? -x2 : x2;
+		int s = 0;
+		
+	for (int i = 0; i < y2; i++) {
+		s = plus(s, y1);	
+	}
+
+	if ((x1 < 0 && x2 > 0) || (x1 > 0 && x2 < 0)) {
+        s = -s;
+	}
+
+	return s;
 	}
 
 	// Returns x^n (for n >= 0)
 	public static int pow(int x, int n) {
-		// Replace the following statement with your code
-		return 0;
+		int y = x;
+		int m = n;
+		int r = 1; 
+		boolean NegativeBase = false;
+
+		if (m == 0) {
+			r = 1;
+		}
+
+		if (x < 0) {
+			y = minus(0, y);
+			NegativeBase = !NegativeBase;
+			
+		}
+			
+		for (int i = 0; i < m; i++) {
+			r = times(r, y);
+		}
+		if (NegativeBase && mod(n, 2) != 0) {
+			return minus(0, r);
+		}
+		
+		return r;
 	}
 
 	// Returns the integer part of x1 / x2 
 	public static int div(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		int y1 = x1;
+		int y2 = x2;
+		int count = 0;
+
+		if (y2 == 0) {
+			throw new IllegalArgumentException("Division by zero is not allowed");
```

Comment:
Exceptions are not in the material taught in class or recitations yet,
This will be covered later in the course. 
However, it wasn't specified on how deal with case of 0.
Therefore, I won't deduct here..